src/bm_sim/calculations.cpp

```diff
@@ -360,6 +360,22 @@ struct csum16 {
   }
 };
 
+struct xor16 {
+  uint16_t operator()(const char *buf, size_t len) const {
+    uint16_t mask = 0x00ff;
+    uint16_t final_xor_value = 0x0000;
+    /* Main loop - 2 bytes at a time */
+    for (unsigned int byte = 0; byte < len; byte += 2) {
+      uint16_t t1 = static_cast<uint16_t>(buf[byte]) << 8;
+      uint16_t t2 = 0x0000;
+      /* Condition returns false on the last byte of an odd-length sequence. */
+      if ((byte + 1) < len) t2 = static_cast<uint16_t>(buf[byte + 1]);
```

Comment:
do we need to have this test in the loop body? There is no need to check whether we are at the end for every iteration.
You could write something like:
```cpp
while (len >= 2) {
  // ...
  len -= 2;
}
if (len > 0) {
  // handle last byte
}
```

Reply:
The loop body has been changed. The check now happens only once after the loop.